docs/training/apis-howto/api-pytorch-trainer-ug.rst

```diff
@@ -0,0 +1,354 @@
+             trainer.fit(
+                 max_length=pytorch.Epoch(1),
+                 checkpoint_period=pytorch.Batch(10),
+                 validation_period=pytorch.Batch(10),
+             )
+
+**To run Trainer API solely on-cluster, the code is much simpler**
+
+.. code::
+
+   def on_cluster():
+       """
+       On-cluster training with Trainer API (entrypoint: python3 train.py)
+       """
+       hparams = det.get_cluster_info().trial.hparams
+
+       with det.pytorch.init() as train_context:
+           trial_inst = model.MNistTrial(train_context, hparams)
+           trainer = det.pytorch.Trainer(trial_inst, train_context)
+           trainer.fit(
+               max_length=pytorch.Epoch(1),
+               checkpoint_period=pytorch.Batch(10),
+               validation_period=pytorch.Batch(10),
+           )
+
+***************************************************
+ Step 5: Submit Your Trial for Training on Cluster
+***************************************************
+
+To run your experiment on cluster, you'll need to create a YAML file. Your YAML file must contain
+searcher configuration and entrypoint.
+
+.. note::
+
+   `global_batch_size` is required if `max_length` is configured in records
+
+.. code::
+
+   name: my_pytorch_trainer_trial
+   hyperparameters:
+     global_batch_size: 32
+   searcher:
+     name: single
+     metric: validation_loss
+     max_length:
+       batches: 937
+   resources:
+     slots_per_trial: 8
+   entrypoint: python3 -m determined.launch.torch_distributed python3 train.py
+
+Submit the trial to the cluster:
+
+.. code:: bash
+
+   det e create det.yaml .
+
+*****************************
+ Step 6: Loading Checkpoints
+*****************************
+
+To load a checkpoint from a checkpoint saved using Trainer, you'll need to download the checkpoint
+to a file directory and use an import helper method to import modules. You should instantiate your
+loaded Trial with a ``CheckpointLoadContext``.
+
+``det.import_from_path`` allows you to import from a specific directory and cleans up afterwards.
+Even if you are importing identically-named files, you can import them as separate modules. This is
+intended to help when you have, for example, a current model_def.py, but also import an older
+model_def.py from a checkpoint into the same interpreter, without conflicts (so long as you import
+them as different names, of course).
+
+``CheckpointLoadContext`` is a special PyTorchTrialContext that can be used to load Trial classes
+outside of normal training loops. It does not support any training features such as metrics
+reporting or uploading checkpoints and is intended for use with the Trainer directly.
+
+.. code::
+
+   import determined as det
+   from determined import pytorch
+   from determined.experimental import client
+       # Download checkpoint and load training code from checkpoint.
+       path = client.get_checkpoint(CHECKPOINT_UUID)
```

Comment:
nit: recommend `MY_CHECKPOINT_UUID`, as the prefix `my_` in code examples usually means "something owned by the user that you need to fill in".  And you're using `my_model_def` and `MyTrial` in other places so it makes sense to do it here too.